### PR TITLE
#122 Occupation search dropdown

### PR DESCRIPTION
**Pull request by rachelchenrc**
Jobseeker can search through and select occupation from the DOL database

Notes: 

- Introduce Material UI Lab [Autocomplete ](https://material-ui.com/components/autocomplete/) for the searchable dropdown

[Trello card](https://trello.com/c/hKy2do57/122-jobseeker-can-search-through-and-select-occupation-from-the-dol-database)

[Live env](https://nj-career-network-dev4.firebaseapp.com/)

Screen Shot 2020-01-27 at 4 38 23 PM

Screen Shot 2020-01-27 at 4 39 16 PM

### CODE REVIEW

File: components/dashboard/Dashboard.js [hunk 1]
```diff
@@ -337,18 +360,25 @@ export default function Dashboard(props) {
...
+                  Learn about the outlook in your area for the occupation that you want.
+                </Typography>
+              </CardContent>
+              <CardActions>
+                <NextLink href="/employment-outlook">
+                  <Button className={classes.exploreButton} fullWidth onClick={handleClick}>
+                    EXPLORE
+                  </Button>
+                </NextLink>
+              </CardActions>
```
Comment (gideonred) [inline components/dashboard/Dashboard.js]: Rachel, didn't have these changes in https://github.com/newjersey/career-network/pull/109/commits? Are you repeating it here to build on top of it? Is there possibly a way separating this PR and PR 109 to be as independent as possible from each other? (basically for this PR, have the home dashboard page be the same and have the user navigate to /employment-outlook/ manually). Let me know if you have questions!

Comment (rachelchenrc) [reply components/dashboard/Dashboard.js]: Great call. yes, it's independent of PR 109. changes removed.

File: scripts/algoliaUpload.js [hunk 2]
```diff
@@ -9,46 +9,63 @@ const { startCase, toLower } = require('lodash/fp');
 
 const CSV_PATH = 'data/Employment Prospects data from NJ.csv';
 const API_KEY = process.env.ALGOLIA_ADMIN_API_KEY;
-const APP_ID = '3XON39SKZ0';
-const INDEX_NAME = 'prod_EMPLOYMENT_PROSPECTS';
+const APP_ID = 'GVXRTXREAI';
```
[the remark this answers is not in the document]
Comment (rachelchenrc) [reply scripts/algoliaUpload.js]: yes. changed back.

File: components/employmentOutlook/AutocompleteDropdown.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+function AutocompleteSearch({ hits, currentRefinement, refine }) {
+  const classes = useStyles();
+  const options = hits.map(option => ({
+    default: `Occupations`,
+    ...option,
+  }));
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (rachelchenrc) [reply components/employmentOutlook/AutocompleteDropdown.js]: removed

**Comment by rachelchenrc** [comment]
> So, not the biggest deal, but…
> 
> Upon page load, this pulls all data into memory from one API call:
> 
> Screen Shot 2020-01-29 at 2 41 36 PM
> 
> User input then triggers additional (redundant) searches:
> 
> Screen Shot 2020-01-29 at 2 41 57 PM
> 
> Would be nice to pick one approach or the other (not both) -- either seems fine for now, but that might change if we 10x the data volume.

@rossdakin Need further thoughts, but it's kind of expected because per design the user should be able to select from all occupations in dropdown before typing anything in the search bar. I'm also concerned about the multiple calls from InstantSearch Autocomplete. but if we move away we'll lose algolia search's benefit like the fuzziness. I may need some extra investigation. but let me know if you have any suggestions. Thanks!

File: components/employmentOutlook/EmploymentOutlook.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    height: 400,
+  },
+}));
+
+function EmploymentOutlook() {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.root}>
+      <NextLink href="/dashboard">
```
[the remark this answers is not in the document]
Comment (gideonred) [reply components/employmentOutlook/EmploymentOutlook.js]: From Crandall:

on #122 - per the code review - we will breakout a ticket so the ‘Return to Dashboard’ element in the sub-nav becomes a global element. we will not have that link within the scope of the current work in implementation.

**Comment by rossdakin** [comment]
> @rossdakin Need further thoughts, but it's kind of expected because per design the user should be able to select from all occupations in dropdown before typing anything in the search bar. I'm also concerned about the multiple calls from InstantSearch Autocomplete. but if we move away we'll lose algolia search's benefit like the fuzziness. I may need some extra investigation. but let me know if you have any suggestions. Thanks!

Great point, @rachelchenrc – looks good to me as is. Thank you!

**Pull request merged**